### PR TITLE
Add support for building noarch: python packages

### PR DESCRIPTION
**Pull request by jjhelmus**
Build noarch: python packages only on the linux-64 platform while testing on all other platforms.

### CODE REVIEW

File: tests/test_execute.py [hunk 1]
```diff
@@ -252,3 +252,43 @@ def test_collapse_with_win_matrix_and_subpackages(monkeypatch):
...
+    assert tasks.node['test-pkg_a-1.0.0-on-win-32']['test_only'] == True
+    # 8 edges
+    # * 6 pkg_b nodes have an edge to the pkg_a build node
+    # * 2 pkg_a tests nodes with edges to the pkg_a build_node
+    print(tasks.edges())
+    assert len(tasks.edges()) == 8
+    a_build_node = 'pkg_a-1.0.0-on-centos5-64'
+    assert ('test-pkg_a-1.0.0-on-osx-109', a_build_node) in tasks.edges()
+    assert ('test-pkg_a-1.0.0-on-win-32', a_build_node) in tasks.edges()
+    assert ('pkg_b-1.0.0-python_3.6-on-osx-109', a_build_node) in tasks.edges()
```
Comment (msarahan) [inline tests/test_execute.py]: Nice.  Should pkg_b stuff also have edges to the outcomes of all the test nodes for a?  I don't think it's strictly necessary.  Might be overconstraining, especially if we have one platform that we just want to ignore.  Perhaps best to keep it as you have it.

File: conda_concourse_ci/compute_build_graph.py [hunk 2]
```diff
@@ -203,9 +203,13 @@ def add_recipe_to_graph(recipe_dir, graph, run, worker, conda_resolve,
             continue
 
         name = package_key(metadata, worker['label'], run)
+        noarch_python = metadata.noarch == 'python'
```
Comment (msarahan) [inline conda_concourse_ci/compute_build_graph.py]: what about noarch: generic?  We should be able to collapse it also, right?

Comment (jjhelmus) [reply conda_concourse_ci/compute_build_graph.py]: I think so but I could find any examples where noarch: generic is used outside of CDT packages.

File: tests/test_execute.py (hunk 1 above)
Comment (jjhelmus) [reply tests/test_execute.py]: I think the current setup is better.  `pkg_b` can start building as soon as `pkg_a` exists.  Adding the logic to ignore "known failing" platforms would probably need to be done at the recipe level which would be awkward.  The pipeline will still show a failure which should be a sign to investigate and verify if this failure is expected or not.

File: conda_concourse_ci/compute_build_graph.py (hunk 2 above)
Comment (msarahan) [reply conda_concourse_ci/compute_build_graph.py]: I believe the plan is to employ it somehow for R stuff.  It also gets used once in a while for data-only packages, like nltk-data

Comment (jjhelmus) [reply conda_concourse_ci/compute_build_graph.py]: Added support for `noarch: generic` and `noarch: True`

**Comment by msarahan** [comment]
This looks good to me.  Do you want me to push a commit for keeping conda 4.5 for now?  I don't think we should try to fix 4.6 support in this PR, but it would be nice to get some less noisy tests for this stuff.

**Comment by jjhelmus** [comment]
I'll push a change to .travis.yaml in a moment to pin to conda 4.5

**Comment by jjhelmus** [comment]
Ah, networkx changed to generators in 2.x, was testing with 1.x locally.

**Pull request merged**